src/containers/ProductPage/ProductPage.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { BsArrowLeft, BsCheckCircle } from "react-icons/bs";
+import { Link, useNavigate } from "react-router-dom";
+import { API_URL } from "../../constants/Constance";
+import productsLogo from "../../assets/logo-white.svg";
+import "./ProductPage.css";
+
+const ProductPage = ({ preview }) => {
+  const navigate = useNavigate();
+  const { productID } = useParams();
+  const [product, setProduct] = useState("");
+
+  useEffect(() => {
+    const fetchProduct = async () => {
```

Comment:
Do not create functions inside of useEffect hook. You can create this function outside and call it in useEffect.